backup_integrator_operator/src/charm.py

```diff
@@ -0,0 +1,85 @@
+#!/usr/bin/env python3
+
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+"""backup-integrator charm."""
+
+import logging
+import pathlib
+import typing
+
+import ops
+from charms.backup_integrator.v0.backup import BackupDynamicRequirer
+
+logger = logging.getLogger(__name__)
+
+
+class BackupIntegratorCharm(ops.CharmBase):
+    """Charm the service."""
+
+    _CHARM_OPT_DIR = pathlib.Path("/opt/backup-integrator-charm")
+
+    def __init__(self, *args: typing.Any):
+        """Construct.
+
+        Args:
+            args: Arguments passed to the CharmBase parent constructor.
+        """
+        super().__init__(*args)
+        self._requirer = BackupDynamicRequirer(charm=self)
+        self.framework.observe(self.on.backup_relation_created, self._reconcile)
+        self.framework.observe(self.on.backup_relation_changed, self._reconcile)
+        self.framework.observe(self.on.config_changed, self._reconcile)
+        self.framework.observe(self.on.upgrade_charm, self._reconcile)
+
+    def _save_script(self, config_option: str) -> pathlib.Path | None:
+        """Save the content of a script configuration.
+
+        Args:
+            config_option: The name of the script configuration option.
+
+        Returns:
+            The path to the saved script if the configuration value is not empty, otherwise, None.
+        """
+        content = typing.cast(str | None, self.config.get(config_option))
+        if not content:
+            return None
+        script_path = (
+            self._CHARM_OPT_DIR / self.unit.name.replace("/", "-") / "scripts" / config_option
+        )
+        script_path.parent.mkdir(parents=True, exist_ok=True)
+        script_path.write_text(content, encoding="utf-8")
+        script_path.chmod(0o755)
+        return script_path
+
+    def _reconcile(self, _: ops.EventBase) -> None:
+        """Reconciles the charm."""
+        fileset_config = typing.cast(str, self.config.get("fileset", "")).strip()
+        if not fileset_config:
+            self.unit.status = ops.BlockedStatus("missing fileset config")
+            return
+        fileset = [file.strip() for file in fileset_config.split(",") if file.strip()]
```

Comment:
Since the library is taking care of that, is it necessary ?

Reply:
It matches the type signature of the `request_backup` function, so it makes linting/type checking a little better.